src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -197,9 +198,9 @@ protected void reloadConfig() {
     }
     try {
       log.info(String.format("Reloading hdfs-site.xml from %s", configUri));
-      String cfgCmd[] = new String[]{"sh", "-c",
-          String.format("curl -o hdfs-site.xml %s ; cp hdfs-site.xml etc/hadoop/", configUri)};
-      Process process = Runtime.getRuntime().exec(cfgCmd);
+      ProcessBuilder processBuilder = new ProcessBuilder("sh", "-c",
+          String.format("curl -o hdfs-site.xml %s ; cp hdfs-site.xml etc/hadoop/", configUri));
+      Process process = processBuilder.start();
       //TODO(nicgrayson) check if the config has changed
       redirectProcess(process);
       int exitCode = process.waitFor();
```

Comment:
This is true, but the reload should be done periodically on a jetty server. Logging an error would be better than killing the executor completely as it may succeed on a retry.